Atril backports handling deprecations and unused code removals

Piiit opened the pull request:
From Atril 1.17.0 until Atril 1.18 inclusive, backport all commits that handle GTK deprecations or remove old code

This PR adds `if GTK_CHECK_VERSION (3, 16, 0)` to some places. Is that needed, or could we also just bump the GTK version to 3.18 as it is in LM 18? I will keep this PR as WIP, until this is answered :-)

JosephMcc commented:
For now we need to support gtk3.14 for LMDE. There is no real reason for the 3_16 version checks. All these do is hide deprecation warnings during build. They are just something else that has to be cleaned up later. That kind of thing can be cleaned up when we bump required gtk versions. Once we hit LMDE3 and Mint19 we can safely move this stuff to gtk3.22 support since it is now stable.

Piiit commented:
@JosephMcc I had a second thought about gtk version checks. I think we should keep this PR as it is, it is easier to spot and remove them later, then to skip a backport commit and find it later on. Currently, we have some gtk version checks for 3.20 as well. What do you think?

Piiit commented:
@JosephMcc @clefebvre I've removed the gtk checks for 3.16 for now, and added `gtk_widget_set_[vh]align` instead of the deprecated `gtk_misc_set_alignment`. Should be ready to merge.
